pandas/core/dtypes/cast.py

```diff
@@ -696,6 +698,8 @@ def astype_nansafe(arr, dtype, copy=True, skipna=False):
         if is_object_dtype(dtype):
             return tslib.ints_to_pydatetime(arr.view(np.int64))
         elif dtype == np.int64:
```

Comment:
this likely needs to be is_integer_dtype(dtype)

Reply:
I think this might actually need to be `np.int64` since above we check for `datetime64` (getting some weird test failures after this change).  Looking at the `numpy` documentation it seems the output of `view` is "unpredictable" when the precision differs between the data types (the shape of the output can even change).